### DISCUSSION TITLE
Best Approach for Result Collection in Autogen GroupChat with Multiple Agents?

### DISCUSSION THREAD

**Discussion by skim0011** [microsoft/autogen#4211] [Q&A]
In an Autogen GroupChat, what's the best approach for creating a "control tower" that can collect and manage results from all agents? For example, if I have data like [{id1}, {id2}, ..., {id10000}] and multiple agents, how can I coordinate the workflow so:

Agent1 checks all data items, picks some (e.g., id1, id2), and applies fixes.
Agent2 also checks all items, picks different ones (e.g., id3, id4), and applies fixes.
How can the GroupChat coordinate between Agent1 and Agent2, then compile a final list of fixed elements [{id1}, {id2}, {id3}, {id4}]?

Thanks

**Comment by ekzhu** [comment]
For this we don't apply group chat, which requires the agent to share all context.

You can implement a map-reduce like pattern using AutoGen Core's API.

**Reply by skim0011** [reply]
thanks for your answer, what if it's ok to be like below?:

Agent1 checks all data items, picks some (e.g., id1, id2), and applies fixes.
Agent2 also checks all items, picks different ones (e.g., id2) - overrides id2 
Agent3 also checks all items, picks different ones (e.g., id3), and applies fixes.

the main question is, if autogen's multiagent can do separate tasks and then give me a summarised/ unionised  result

**Reply by ekzhu** [reply]
Sounds like an application of mixture of agents? 

https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/design-patterns/mixture-of-agents.html

In your case a single layer is enough.
